src/base/bittorrent/session.cpp

```diff
@@ -2349,10 +2349,10 @@ void Session::exportTorrentFile(TorrentHandle *const torrent, TorrentExportFolde
 void Session::generateResumeData(bool final)
 {
     foreach (TorrentHandle *const torrent, m_torrents) {
-        if (!torrent->isValid()) continue;
-        if (torrent->hasMissingFiles()) continue;
-        if (torrent->isChecking() || torrent->hasError()) continue;
+        if (torrent->isChecking() || torrent->isPaused()) continue;
         if (!final && !torrent->needSaveResumeData()) continue;
+        if (torrent->hasMissingFiles() || torrent->hasError()) continue;
+        if (!torrent->isValid()) continue;
```

Comment:
maybe still a good idea to have this on top? I would prefer knowing the handle is valid before checking other status.

Reply:
You're right. Done.

`isValid()` can return `true` here for some reason. Although this is the most unlikely case, it would be better to have it at the top, otherwise we may get an exception in one of the following methods.

Reply:
Still undone.

Reply:
Oops... forgot to push it.
Fixed now, I hope :)